### ISSUE TITLE
Manualy change the sorting of dataproviders

### ISSUE THREAD

**Issue by yariksav** [yiisoft/yii2#10243]
If i need to set sorting manualy - i forced to apply hack: $_GET['sort'] = 'name-,author' etc, because i've found only this way in Sort class. Perhaps you'll do another way in yii/data/Sort class, like Sort::sort = 'name-,author' or something like ->sort(['name'=>SORT_DESC])
Thanks

**Comment by cbhp** [comment]
I have 3 working variants:

1.) 
``` 
        $searchModel = new MemberSearch();
        $dataProvider = $searchModel->search(Yii::$app->request->queryParams);
        $dataProvider->sort->enableMultiSort = true;
        $dataProvider->sort->defaultOrder = [
            'lastname' => SORT_ASC,
            'firstname' => SORT_ASC,
        ];
 ``` 

2.) 
``` 
        $dataProvider = new ActiveDataProvider([
            'query' => Member::find()),
            'sort' => [
                'defaultOrder' => [
                    'lastname' => SORT_ASC,
                    'firstname' => SORT_ASC,
                ]
            ]
        ]);
 ``` 

3.) 
``` 
        $dataProvider = new ActiveDataProvider([
            'query' => Member::find()->orderBy(['lastvisit' => SORT_DESC]),
        	'sort' => false,
        ]);
 ```

**Comment by yariksav** [comment]
Thanks For answer. I use my own GrivView, and there sorting change AFTER DataProvider has initialized. I already have object $dataProvider (before calling getModels()) initialized in other module, and I automaticaly calculate pagination, filters and sotring.
That why your variant do not approach for me.
I already have used another hack:
$this->dataProvider->query->addOrderBy(new Expression($key . ' ' . $value));
but if I have initialized default column - sotring in this colum not work.

Thats why i've desided to use hack $_GET['sort'] = 'name-,author', but it is not beautiful code))

**Comment by klimov-paul** [comment]
Resolved by commit 90b9ed87bd0265b2a645c6828635ab922fa7fd7c
